nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java

```diff
@@ -136,10 +150,9 @@ public Comment getExternalComment(CoreSession session, String entityId) {
         DocumentModel commentModel = getExternalCommentModel(session, entityId);
         // Check the read permissions using the given session
         checkReadCommentPermissions(session, commentModel.getRef());
-        return Framework.doPrivileged(() -> COMMENT_DOC_TYPE.equals(commentModel.getType())
-               ? Comments.newComment(commentModel)
-               : Comments.newAnnotation(commentModel)
-        );
+        return Framework.doPrivileged(
+                () -> COMMENT_DOC_TYPE.equals(commentModel.getType()) ? Comments.newComment(commentModel)
```

Comment:
We have more and more such instantiations, sometimes we don't do this ternary (in previous code), which leads to incomprehension.

What do you think to move this to the `Comments` class under a method like `create`?

Reply:
yep, as discussed i was tempted to go further on refactoring the `AnnotationService` some works was done on the previous Comments ticket (the rework / security one). What i propose here is to delegate more things into `CommentManager` i stared doing it see  [my commit](https://github.com/nuxeo/nuxeo/pull/3549/commits/0bc6c51dd3e5d6c0ec1c59d090712801969a5674) this is a proposal i want to go more further to deprecate some code in the `Comments` class and expose just what we want not the whole logic, we can discuss about it on one to one ;)

thx